In [73]:
#1. Perform pre-processing and visualization of the dataset. Split the data into train and test sets. Also
#identify the useful columns and drop the unnecessary ones
import pandas as pd
import numpy as np
from sklearn.preprocessing import LabelEncoder
from sklearn.model_selection import train_test_split as tts
import matplotlib as plt

In [69]:
#analyzing and preprocessing the data
data = pd.read_csv("/content/titanic-titanic.csv")
data = data.apply(pd.to_numeric, errors = "coerce").fillna(data)
print(data)

     PassengerId                                               Name  Pclass  \
0              1                            Braund, Mr. Owen Harris       3   
1              2  Cumings, Mrs. John Bradley (Florence Briggs Th...       1   
2              3                             Heikkinen, Miss. Laina       3   
3              4       Futrelle, Mrs. Jacques Heath (Lily May Peel)       1   
4              5                           Allen, Mr. William Henry       3   
..           ...                                                ...     ...   
886          887                              Montvila, Rev. Juozas       2   
887          888                       Graham, Miss. Margaret Edith       1   
888          889           Johnston, Miss. Catherine Helen "Carrie"       3   
889          890                              Behr, Mr. Karl Howell       1   
890          891                                Dooley, Mr. Patrick       3   

        Sex   Age            Ticket     Fare Cabin 

In [70]:
print(data.isnull().any())
print(data.Age.isnull().sum())
print(data.Embarked.isnull().sum())
print(data.Cabin.isnull().sum())
col_n = data.iloc[0]
for i in col_n:
  print(type(i))

PassengerId    False
Name           False
Pclass         False
Sex            False
Age             True
Ticket         False
Fare           False
Cabin           True
Embarked        True
Survived       False
dtype: bool
177
2
687
<class 'numpy.int64'>
<class 'str'>
<class 'numpy.int64'>
<class 'str'>
<class 'float'>
<class 'str'>
<class 'float'>
<class 'float'>
<class 'str'>
<class 'numpy.int64'>


Since the cabin column has 687 out of 891 which is 77.104% of the entries missing we can safely drop this column from our dataset without any major reprucussions.

Simiarly we can also drop names since all of them are different and thus the complete column is noisy and would adversely affect our dataset.

We can also drop passenger Id as it is just a column which gives us serial no. which we already have from loading our dataset.

Similarly we can drop ticket as we already have seperate passanger class.

In [71]:
data.drop("Cabin",axis = 'columns',inplace = True)
data.drop("Name",axis = 'columns',inplace = True)
data.drop("PassengerId",axis = 'columns',inplace = True)
data.drop("Ticket",axis = 'columns',inplace = True)

In [72]:
col_n = list(data.columns)
data.Embarked.fillna(0,inplace = True)
rows_dropped = []
for i,row in data.iterrows():
  if row['Embarked'] == 0:
    rows_dropped.append(i)
data = data.drop(list(set(rows_dropped)), axis=0, inplace=False)

In [74]:
print(rows_dropped)

[61, 829]


In [75]:
data['Age'] = data.Age.fillna(data.Age.median())

In [76]:
label_encode = LabelEncoder()
labels_sex = label_encode.fit_transform(data.Sex)
data["Sex"] = labels_sex

In [77]:
label_encode1 = LabelEncoder()
labels_embarked = label_encode.fit_transform(data.Embarked)
data["Embarked"] = labels_embarked

Male ---> 1
Female ---> 0
Embarked: S ---> 2
Embarked: Q ---> 1
Embarked: C ---> 0